# Titanic Survival

#### Grading:


- Code: 90 pts
- Markdown Documentation: 10 pts


We are going to study the survival rate of passengers on titanic and what variables affected survival.

Load the dataset in `titanic.xls`. It contains data on all the passengers that travelled on the Titanic.

In [63]:
from IPython.core.display import HTML
HTML(filename='../data/titanic.html')

In [64]:
import pandas as pd

In [65]:
# you would need xlrd - pip install xlrd
t_file = pd.ExcelFile('../data/titanic.xls')
t_df = t_file.parse("titanic", header=None)
t_df.head()

0         1                                     2       3       4   \
0  pclass  survived                                  name     sex     age   
1       1         1         Allen, Miss. Elisabeth Walton  female      29   
2       1         1        Allison, Master. Hudson Trevor    male  0.9167   
3       1         0          Allison, Miss. Helen Loraine  female       2   
4       1         0  Allison, Mr. Hudson Joshua Creighton    male      30   

      5      6       7         8        9         10    11    12  \
0  sibsp  parch  ticket      fare    cabin  embarked  boat  body   
1      0      0   24160  211.3375       B5         S     2   NaN   
2      1      2  113781    151.55  C22 C26         S    11   NaN   
3      1      2  113781    151.55  C22 C26         S   NaN   NaN   
4      1      2  113781    151.55  C22 C26         S   NaN   135   

                                13  
0                        home.dest  
1                     St Louis, MO  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [66]:
new_header = t_df.iloc[0] 
t_df = t_df[1:] 
t_df.columns = new_header

In [67]:
t_df.head()

0 pclass survived                                             name     sex  \
1      1        1                    Allen, Miss. Elisabeth Walton  female   
2      1        1                   Allison, Master. Hudson Trevor    male   
3      1        0                     Allison, Miss. Helen Loraine  female   
4      1        0             Allison, Mr. Hudson Joshua Creighton    male   
5      1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

0     age sibsp parch  ticket      fare    cabin embarked boat body  \
1      29     0     0   24160  211.3375       B5        S    2  NaN   
2  0.9167     1     2  113781    151.55  C22 C26        S   11  NaN   
3       2     1     2  113781    151.55  C22 C26        S  NaN  NaN   
4      30     1     2  113781    151.55  C22 C26        S  NaN  135   
5      25     1     2  113781    151.55  C22 C26        S  NaN  NaN   

0                        home.dest  
1                     St Louis, MO  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON  
5  Montreal, PQ / Chesterville, ON

### Women and children first?

*** 1. Use the `groupby` method to calculate the proportion of passengers that survived by sex. (25 pts)*** 

Lets Calculate total passengers by sex

In [68]:
t_df.groupby('sex').size()

sex
female    466
male      843
dtype: int64

Total male passengers are 843 and total female passengers are 466

In [69]:
male_passenger_count = t_df.groupby('sex').size()[1]
female_passenger_count = t_df.groupby('sex').size()[0]

Using groupby to find out total survived passengers

In [70]:
Survived = t_df.groupby('survived')
Survived.size()

survived
0    809
1    500
dtype: int64

Total Passengers Survived are 500

In [71]:
Gender_groupby = Survived.get_group(1).groupby('sex')
Gender_groupby.size()

sex
female    339
male      161
dtype: int64

In [72]:
male_survived = Gender_groupby.size()[1]
female_survived = Gender_groupby.size()[0]

Total Female survivors are 339 and Total Male survivors are 161

#### Proportion of passengers survived

In [73]:
#survival percentage of Female passengers
survival_percentage_female = (female_survived/female_passenger_count)*100
survival_percentage_male = (male_survived/male_passenger_count)*100

In [74]:
print("Percentage of female passengers survived : ", round(survival_percentage_female,3),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male,3),"%")

Percentage of female passengers survived :  72.747 %
Percentage of male passengers survived :  19.098 %


*** 2. Calculate the same proportion, but by class and sex. (25 pts)*** 

As we have seen in the documentation above, there are three classes. i.e., 1,2 and 3

Let us sort by class, to find class wise stats 

In [75]:
Class_groupby = t_df.groupby('pclass')
Class_groupby.head(5)

0   pclass survived                                             name     sex  \
1        1        1                    Allen, Miss. Elisabeth Walton  female   
2        1        1                   Allison, Master. Hudson Trevor    male   
3        1        0                     Allison, Miss. Helen Loraine  female   
4        1        0             Allison, Mr. Hudson Joshua Creighton    male   
5        1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   
324      2        0                              Abelson, Mr. Samuel    male   
325      2        1            Abelson, Mrs. Samuel (Hannah Wizosky)  female   
326      2        0                   Aldworth, Mr. Charles Augustus    male   
327      2        0                       Andrew, Mr. Edgardo Samuel    male   
328      2        0                         Andrew, Mr. Frank Thomas    male   
601      3        0                              Abbing, Mr. Anthony    male   
602      3        0                    Abbott, Master. Eugene Joseph    male   
603      3        0                      Abbott, Mr. Rossmore Edward    male   
604      3        1                 Abbott, Mrs. Stanton (Rosa Hunt)  female   
605      3        1                      Abelseth, Miss. Karen Marie  female   

0       age sibsp parch      ticket      fare    cabin embarked boat body  \
1        29     0     0       24160  211.3375       B5        S    2  NaN   
2    0.9167     1     2      113781    151.55  C22 C26        S   11  NaN   
3         2     1     2      113781    151.55  C22 C26        S  NaN  NaN   
4        30     1     2      113781    151.55  C22 C26        S  NaN  135   
5        25     1     2      113781    151.55  C22 C26        S  NaN  NaN   
324      30     1     0   P/PP 3381        24      NaN        C  NaN  NaN   
325      28     1     0   P/PP 3381        24      NaN        C   10  NaN   
326      30     0     0      248744        13      NaN        S  NaN  NaN   
327      18     0     0      231945      11.5      NaN        S  NaN  NaN   
328      25     0     0  C.A. 34050      10.5      NaN        S  NaN  NaN   
601      42     0     0   C.A. 5547      7.55      NaN        S  NaN  NaN   
602      13     0     2   C.A. 2673     20.25      NaN        S  NaN  NaN   
603      16     1     1   C.A. 2673     20.25      NaN        S  NaN  190   
604      35     1     1   C.A. 2673     20.25      NaN        S    A  NaN   
605      16     0     0      348125      7.65      NaN        S   16  NaN   

0                                   home.dest  
1                                St Louis, MO  
2             Montreal, PQ / Chesterville, ON  
3             Montreal, PQ / Chesterville, ON  
4             Montreal, PQ / Chesterville, ON  
5             Montreal, PQ / Chesterville, ON  
324                       Russia New York, NY  
325                       Russia New York, NY  
326                        Bryn Mawr, PA, USA  
327  Buenos Aires, Argentina / New Jersey, NJ  
328            Cornwall, England Houghton, MI  
601                                       NaN  
602                       East Providence, RI  
603                       East Providence, RI  
604                       East Providence, RI  
605                    Norway Los Angeles, CA

#### Class 1 

In [76]:
Class_1 = Class_groupby.get_group(1)
Class_1.head()

0 pclass survived                                             name     sex  \
1      1        1                    Allen, Miss. Elisabeth Walton  female   
2      1        1                   Allison, Master. Hudson Trevor    male   
3      1        0                     Allison, Miss. Helen Loraine  female   
4      1        0             Allison, Mr. Hudson Joshua Creighton    male   
5      1        0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

0     age sibsp parch  ticket      fare    cabin embarked boat body  \
1      29     0     0   24160  211.3375       B5        S    2  NaN   
2  0.9167     1     2  113781    151.55  C22 C26        S   11  NaN   
3       2     1     2  113781    151.55  C22 C26        S  NaN  NaN   
4      30     1     2  113781    151.55  C22 C26        S  NaN  135   
5      25     1     2  113781    151.55  C22 C26        S  NaN  NaN   

0                        home.dest  
1                     St Louis, MO  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON  
5  Montreal, PQ / Chesterville, ON

In [77]:
print("Class 1 total passengers")
Class_1.groupby('sex').size()

Class 1 total passengers


sex
female    144
male      179
dtype: int64

In [78]:
Class_1_total_female = Class_1.groupby('sex').size()[0]
Class_1_total_male = Class_1.groupby('sex').size()[1]

In [79]:
Class_1_survived = Class_1.groupby('survived').get_group(1)

In [80]:
Class_1_survived.to_csv('../data/Class_1_survived.csv',index=False)

In [81]:
Class_1_survived_sex = Class_1_survived.groupby('sex')

In [82]:
print("Class 1 total survivors")
Class_1_survived_sex.size()

Class 1 total survivors


sex
female    139
male       61
dtype: int64

In [83]:
Class_1_survived_female = Class_1_survived_sex.size()[0]
Class_1_survived_male = Class_1_survived_sex.size()[1]

#### Class 1 survival stats

In [84]:
#survival percentage of Female passengers
survival_percentage_female = (Class_1_survived_female/Class_1_total_female)*100
survival_percentage_male = (Class_1_survived_male/Class_1_total_male)*100
print("Percentage of female passengers survived in Class 1 : ", round(survival_percentage_female,2),"%")
print("Percentage of male passengers survived in Class 1: ", round(survival_percentage_male,2),"%")

Percentage of female passengers survived in Class 1 :  96.53 %
Percentage of male passengers survived in Class 1:  34.08 %


Class 2

In [85]:
Class_2 = Class_groupby.get_group(2)
print("Class 2 total passengers")
Class_2.groupby('sex').size()

Class 2 total passengers


sex
female    106
male      171
dtype: int64

In [86]:
Class_2_total_female = Class_2.groupby('sex').size()[0]
Class_2_total_male = Class_2.groupby('sex').size()[1]
Class_2_survived = Class_2.groupby('survived').get_group(1)
Class_2_survived.to_csv('../data/Class_2_survived.csv',index=False)
Class_2_survived_sex = Class_2_survived.groupby('sex')
print("Class 2 total survivors")
Class_2_survived_sex.size()

Class 2 total survivors


sex
female    94
male      25
dtype: int64

In [87]:
Class_2_survived_female = Class_2_survived_sex.size()[0]
Class_2_survived_male = Class_2_survived_sex.size()[1]

#### Class 2 survival stats

In [88]:
#survival percentage of Female passengers
survival_percentage_female = (Class_2_survived_female/Class_2_total_female)*100
survival_percentage_male = (Class_2_survived_male/Class_2_total_male)*100
print("Percentage of female passengers survived in Class 2 : ", round(survival_percentage_female,2),"%")
print("Percentage of male passengers survived in Class 2: ", round(survival_percentage_male,2),"%")

Percentage of female passengers survived in Class 2 :  88.68 %
Percentage of male passengers survived in Class 2:  14.62 %


Class 3

In [89]:
Class_3 = Class_groupby.get_group(3)
print("Class 3 total passengers")
Class_3.groupby('sex').size()

Class 3 total passengers


sex
female    216
male      493
dtype: int64

In [90]:
Class_3_total_female = Class_3.groupby('sex').size()[0]
Class_3_total_male = Class_3.groupby('sex').size()[1]
Class_3_survived = Class_3.groupby('survived').get_group(1)
Class_3_survived.to_csv('../data/Class_3_survived.csv',index=False)
Class_3_survived_sex = Class_3_survived.groupby('sex')
print("Class 3 total survivors")
Class_3_survived_sex.size()

Class 3 total survivors


sex
female    106
male       75
dtype: int64

In [91]:
Class_3_survived_female = Class_3_survived_sex.size()[0]
Class_3_survived_male = Class_3_survived_sex.size()[1]

In [92]:
#survival percentage of Female passengers
survival_percentage_female = (Class_3_survived_female/Class_3_total_female)*100
survival_percentage_male = (Class_3_survived_male/Class_3_total_male)*100
print("Percentage of female passengers survived in Class 3 : ", round(survival_percentage_female,2),"%")
print("Percentage of male passengers survived in Class 3: ", round(survival_percentage_male,2),"%")

Percentage of female passengers survived in Class 3 :  49.07 %
Percentage of male passengers survived in Class 3:  15.21 %


*** 3. Create age categories: children (under 14 years), adolescents (14-20), adult (21-64), and senior(65+), and calculate survival proportions by age category, class and sex. (40 pts)***

In [93]:
#creating masks for each age group
children_mask = (t_df.age < 14)
adolescent_mask = (t_df.age >= 14) & (t_df.age <= 20)
adult_mask = (t_df.age >= 21) & (t_df.age <= 64)
senior_mask = (t_df.age >= 65)

In [94]:
#creating separate dataframes for each age group
children_df = t_df[children_mask]
adolescent_df = t_df[adolescent_mask]
adult_df = t_df[adult_mask]
senior_df = t_df[senior_mask]

#### Children 

In [95]:
Class_groupby = children_df.groupby('pclass')

In [96]:
# Class 1
Class_1 = Class_groupby.get_group(1)
print("Class 1 total passengers")
Class_1.groupby('sex').size()

Class 1 total passengers


sex
female    1
male      5
dtype: int64

In [97]:
Class_1_total_female = Class_1.groupby('sex').size()[0]
Class_1_total_male = Class_1.groupby('sex').size()[1]

In [98]:
Class_1_survived = Class_1.groupby('survived').get_group(1)
children_class_1_survived_sex = Class_1_survived.groupby('sex')
print("Class 1 total survivors")
children_class_1_survived_sex.size()

Class 1 total survivors


sex
male    5
dtype: int64

In class 1, there are 5 male survivors and 0 female survivors

In [99]:
Class_1_total_female_survivors = 0
Class_1_total_male_survivors = children_class_1_survived_sex.size()[0]

In [100]:
# Calculating children class 1 stats
survival_percentage_female_class1_children = (Class_1_total_female_survivors/Class_1_total_female)*100
survival_percentage_male_class1_children = (Class_1_total_male_survivors/Class_1_total_male)*100

In [101]:
#Class 2
Class_2 = Class_groupby.get_group(2)
print("Class 2 total passengers")
Class_2.groupby('sex').size()

Class 2 total passengers


sex
female    14
male      11
dtype: int64

In [102]:
Class_2_total_female = Class_2.groupby('sex').size()[0]
Class_2_total_male = Class_2.groupby('sex').size()[1]
Class_2_survived = Class_2.groupby('survived').get_group(1)
Class_2_survived_sex = Class_2_survived.groupby('sex')
print("Class 2 total survivors")
Class_2_survived_sex.size()

Class 2 total survivors


sex
female    14
male      11
dtype: int64

We can see that all of the passsengers survived in class 2

In [103]:
Class_2_survived_female = Class_2_survived_sex.size()[0]
Class_2_survived_male = Class_2_survived_sex.size()[1]

In [104]:
survival_percentage_female_class2_children = (Class_2_survived_female/Class_2_total_female)*100
survival_percentage_male_class2_children = (Class_2_survived_male/Class_2_total_male)*100

In [105]:
#Class 3
Class_3 = Class_groupby.get_group(3)
print("Class 3 total passengers")
Class_3.groupby('sex').size()

Class 3 total passengers


sex
female    31
male      37
dtype: int64

In [106]:
Class_3_total_female = Class_3.groupby('sex').size()[0]
Class_3_total_male = Class_3.groupby('sex').size()[1]
Class_3_survived = Class_3.groupby('survived').get_group(1)
Class_3_survived_sex = Class_3_survived.groupby('sex')
print("Class 3 total survivors")
Class_3_survived_sex.size()

Class 3 total survivors


sex
female    15
male      12
dtype: int64

In [107]:
Class_3_survived_female = Class_3_survived_sex.size()[0]
Class_3_survived_male = Class_3_survived_sex.size()[1]

In [108]:
survival_percentage_female_class3_children = (Class_3_survived_female/Class_3_total_female)*100
survival_percentage_male_class3_children = (Class_3_survived_male/Class_3_total_male)*100

Adolescent

In [109]:
Class_groupby = adolescent_df.groupby('pclass')

In [110]:
#class 1
Class_1 = Class_groupby.get_group(1)
print("Class 1 total passengers")
print(Class_1.groupby('sex').size())
Class_1_total_female = Class_1.groupby('sex').size()[0]
Class_1_total_male = Class_1.groupby('sex').size()[1]
Class_1_survived = Class_1.groupby('survived').get_group(1)
children_class_1_survived_sex = Class_1_survived.groupby('sex')
print("Class 1 total survivors")
print(children_class_1_survived_sex.size())
Class_1_total_female_survivors = children_class_1_survived_sex.size()[0]
Class_1_total_male_survivors = children_class_1_survived_sex.size()[1]
survival_percentage_female_class1_adolescent = (Class_1_total_female_survivors/Class_1_total_female)*100
survival_percentage_male_class1_adolescent = (Class_1_total_male_survivors/Class_1_total_male)*100

Class 1 total passengers
sex
female    15
male       5
dtype: int64
Class 1 total survivors
sex
female    15
male       1
dtype: int64


In [111]:
# Class 2
Class_2 = Class_groupby.get_group(2)
print("Class 2 total passengers")
print(Class_2.groupby('sex').size())
Class_2_total_female = Class_2.groupby('sex').size()[0]
Class_2_total_male = Class_2.groupby('sex').size()[1]
Class_2_survived = Class_2.groupby('survived').get_group(1)
children_class_2_survived_sex = Class_2_survived.groupby('sex')
print("Class 2 total survivors")
print(children_class_2_survived_sex.size())
Class_2_total_female_survivors = children_class_2_survived_sex.size()[0]
Class_2_total_male_survivors = children_class_2_survived_sex.size()[1]
survival_percentage_female_class2_adolescent = (Class_2_total_female_survivors/Class_2_total_female)*100
survival_percentage_male_class2_adolescent = (Class_2_total_male_survivors/Class_2_total_male)*100


Class 2 total passengers
sex
female    13
male      17
dtype: int64
Class 2 total survivors
sex
female    12
male       2
dtype: int64


In [112]:
# Class 3
Class_3 = Class_groupby.get_group(3)
print("Class 3 total passengers")
print(Class_3.groupby('sex').size())
Class_3_total_female = Class_3.groupby('sex').size()[0]
Class_3_total_male = Class_3.groupby('sex').size()[1]
Class_3_survived = Class_3.groupby('survived').get_group(1)
children_class_3_survived_sex = Class_3_survived.groupby('sex')
print("Class 3 total survivors")
print(children_class_3_survived_sex.size())
Class_3_total_female_survivors = children_class_3_survived_sex.size()[0]
Class_3_total_male_survivors = children_class_3_survived_sex.size()[1]
survival_percentage_female_class3_adolescent = (Class_3_total_female_survivors/Class_3_total_female)*100
survival_percentage_male_class3_adolescent = (Class_3_total_male_survivors/Class_3_total_male)*100


Class 3 total passengers
sex
female    35
male      64
dtype: int64
Class 3 total survivors
sex
female    19
male       8
dtype: int64


Adults

In [113]:
Class_groupby = adult_df.groupby('pclass')

In [114]:
#class 1
Class_1 = Class_groupby.get_group(1)
print("Class 1 total passengers")
print(Class_1.groupby('sex').size())
Class_1_total_female = Class_1.groupby('sex').size()[0]
Class_1_total_male = Class_1.groupby('sex').size()[1]
Class_1_survived = Class_1.groupby('survived').get_group(1)
children_class_1_survived_sex = Class_1_survived.groupby('sex')
print("Class 1 total survivors")
print(children_class_1_survived_sex.size())
Class_1_total_female_survivors = children_class_1_survived_sex.size()[0]
Class_1_total_male_survivors = children_class_1_survived_sex.size()[1]
survival_percentage_female_class1_adult = (Class_1_total_female_survivors/Class_1_total_female)*100
survival_percentage_male_class1_adult = (Class_1_total_male_survivors/Class_1_total_male)*100

Class 1 total passengers
sex
female    116
male      134
dtype: int64
Class 1 total survivors
sex
female    112
male       46
dtype: int64


In [115]:
# Class 2
Class_2 = Class_groupby.get_group(2)
print("Class 2 total passengers")
print(Class_2.groupby('sex').size())
Class_2_total_female = Class_2.groupby('sex').size()[0]
Class_2_total_male = Class_2.groupby('sex').size()[1]
Class_2_survived = Class_2.groupby('survived').get_group(1)
children_class_2_survived_sex = Class_2_survived.groupby('sex')
print("Class 2 total survivors")
print(children_class_2_survived_sex.size())
Class_2_total_female_survivors = children_class_2_survived_sex.size()[0]
Class_2_total_male_survivors = children_class_2_survived_sex.size()[1]
survival_percentage_female_class2_adult = (Class_2_total_female_survivors/Class_2_total_female)*100
survival_percentage_male_class2_adult = (Class_2_total_male_survivors/Class_2_total_male)*100


Class 2 total passengers
sex
female     76
male      128
dtype: int64
Class 2 total survivors
sex
female    66
male      10
dtype: int64


In [116]:
# Class 3
Class_3 = Class_groupby.get_group(3)
print("Class 3 total passengers")
print(Class_3.groupby('sex').size())
Class_3_total_female = Class_3.groupby('sex').size()[0]
Class_3_total_male = Class_3.groupby('sex').size()[1]
Class_3_survived = Class_3.groupby('survived').get_group(1)
children_class_3_survived_sex = Class_3_survived.groupby('sex')
print("Class 3 total survivors")
print(children_class_3_survived_sex.size())
Class_3_total_female_survivors = children_class_3_survived_sex.size()[0]
Class_3_total_male_survivors = children_class_3_survived_sex.size()[1]
survival_percentage_female_class3_adult = (Class_3_total_female_survivors/Class_3_total_female)*100
survival_percentage_male_class3_adult = (Class_3_total_male_survivors/Class_3_total_male)*100

Class 3 total passengers
sex
female     86
male      244
dtype: int64
Class 3 total survivors
sex
female    38
male      39
dtype: int64


Senior

In [117]:
Class_groupby = senior_df.groupby('pclass')

In [118]:
#class 1
Class_1 = Class_groupby.get_group(1)
print("Class 1 total passengers")
print(Class_1.groupby('sex').size())
Class_1_total_female = Class_1.groupby('sex').size()[0]
Class_1_total_male = Class_1.groupby('sex').size()[1]
Class_1_survived = Class_1.groupby('survived').get_group(1)
children_class_1_survived_sex = Class_1_survived.groupby('sex')
print("Class 1 total survivors")
print(children_class_1_survived_sex.size())
Class_1_total_female_survivors = children_class_1_survived_sex.size()[0]
Class_1_total_male_survivors = children_class_1_survived_sex.size()[1]
survival_percentage_female_class1_senior = (Class_1_total_female_survivors/Class_1_total_female)*100
survival_percentage_male_class1_senior = (Class_1_total_male_survivors/Class_1_total_male)*100

Class 1 total passengers
sex
female    1
male      7
dtype: int64
Class 1 total survivors
sex
female    1
male      1
dtype: int64


In [119]:
# Class 2
Class_2 = Class_groupby.get_group(2)
print("Class 2 total passengers")
Class_2_total_female = Class_2.loc[Class_2['sex']=='female']
Class_2_total_male = Class_2.loc[Class_2['sex']=='male']
print("Total Female: ", Class_2_total_female.shape[0])
print("Total male: ", Class_2_total_male.shape[0])
Class_2_survived = Class_2[Class_2['survived'] == 1]
print("Class 2 total survivors")
Class_2_total_female_survivors = Class_2_survived.loc[Class_2_survived['sex']=='female']
Class_2_total_male_survivors = Class_2_survived.loc[Class_2_survived['sex']=='male']
print("Total Female: ", Class_2_total_female_survivors.shape[0])
print("Total male: ", Class_2_total_male_survivors.shape[0])

#As survivors are zero
survival_percentage_female_class2_senior = 0
survival_percentage_male_class2_senior = 0


Class 2 total passengers
Total Female:  0
Total male:  2
Class 2 total survivors
Total Female:  0
Total male:  0


In [120]:
# Class 3
Class_3 = Class_groupby.get_group(3)
print("Class 3 total passengers")
Class_3_total_female = Class_3.loc[Class_3['sex']=='female']
Class_3_total_male = Class_3.loc[Class_3['sex']=='male']
print("Total Female: ", Class_3_total_female.shape[0])
print("Total male: ", Class_3_total_male.shape[0])
Class_3_survived = Class_3[Class_3['survived'] == 1]
print("Class 3 total survivors")
Class_3_total_female_survivors = Class_3_survived.loc[Class_3_survived['sex']=='female']
Class_3_total_male_survivors = Class_3_survived.loc[Class_3_survived['sex']=='male']
print("Total Female: ", Class_3_total_female_survivors.shape[0])
print("Total male: ", Class_3_total_male_survivors.shape[0])

#As survivors are zero
survival_percentage_female_class3_senior = 0
survival_percentage_male_class3_senior = 0


Class 3 total passengers
Total Female:  0
Total male:  3
Class 3 total survivors
Total Female:  0
Total male:  0


## Children Stats

In [121]:
print("Class 1")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class1_children,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class1_children,2),"%")
print("\n")
print("Class 2")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class2_children,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class2_children,2),"%")
print("\n")
print("Class 3")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class3_children,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class3_children,2),"%")

Class 1
Percentage of female passengers survived :  0.0 %
Percentage of male passengers survived :  100.0 %


Class 2
Percentage of female passengers survived :  100.0 %
Percentage of male passengers survived :  100.0 %


Class 3
Percentage of female passengers survived :  48.39 %
Percentage of male passengers survived :  32.43 %


## Adolescent Stats

In [122]:
print("Class 1")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class1_adolescent,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class1_adolescent,2),"%")
print("\n")
print("Class 2")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class2_adolescent,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class2_adolescent,2),"%")
print("\n")
print("Class 3")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class3_adolescent,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class3_adolescent,2),"%")

Class 1
Percentage of female passengers survived :  100.0 %
Percentage of male passengers survived :  20.0 %


Class 2
Percentage of female passengers survived :  92.31 %
Percentage of male passengers survived :  11.76 %


Class 3
Percentage of female passengers survived :  54.29 %
Percentage of male passengers survived :  12.5 %


## Adults Stats

In [123]:
print("Class 1")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class1_adult,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class1_adult,2),"%")
print("\n")
print("Class 2")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class2_adult,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class2_adult,2),"%")
print("\n")
print("Class 3")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class3_adult,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class3_adult,2),"%")

Class 1
Percentage of female passengers survived :  96.55 %
Percentage of male passengers survived :  34.33 %


Class 2
Percentage of female passengers survived :  86.84 %
Percentage of male passengers survived :  7.81 %


Class 3
Percentage of female passengers survived :  44.19 %
Percentage of male passengers survived :  15.98 %


## Senior Stats

In [124]:
print("Class 1")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class1_senior,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class1_senior,2),"%")
print("\n")
print("Class 2")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class2_senior,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class2_senior,2),"%")
print("\n")
print("Class 3")
print("Percentage of female passengers survived : ", round(survival_percentage_female_class3_senior,2),"%")
print("Percentage of male passengers survived : ", round(survival_percentage_male_class3_senior,2),"%")

Class 1
Percentage of female passengers survived :  100.0 %
Percentage of male passengers survived :  14.29 %


Class 2
Percentage of female passengers survived :  0 %
Percentage of male passengers survived :  0 %


Class 3
Percentage of female passengers survived :  0 %
Percentage of male passengers survived :  0 %
